### DISCUSSION TITLE
Connect height_acquire and geo_process

### DISCUSSION THREAD

**Discussion by loriwon-uw** [UndefiedCse/Heat-Island#60] [General]
It seems like there are two very similar (identical?) functions in the two modules. Are we going to combine them so that one imports from the other? If so, I think it makes sense to make the smaller geo_process the main for that, and then import into height_acquire.

Also, @LilacHo I'm currently importing the create_hexagon from geo_process. Mine has 3 parameters, but your function only seems to take 2. Will this be updated? Or, do I need to call from the height_acquire instead?

**Comment by LilacHo** [comment]
Hi @loriwon-uw  thank you for your work. There is some inconsistence between my work and Zhao's. I am working on this today. In terms of how to call it, I will offer a chunk of code by the end of today. The three parameters you need are lat, lon, and radius?
BTW. I haven't finished geo_process.py. I am gonna deny your change in the pull request. (I don't know how at this point and I may try out - so I apologize if something weird happens on your end.)

**Comment by LilacHo** [comment]
Here is my output:
{'centroid_stat_total_height_area': 3.6256375409780256e-06, 'centroid_stat_avg_height_area': 1.3976175047616683, 'centroid_stat_mean': 6.755335578098756, 'centroid_stat_median': 6.49275506370837, 'centroid_stat_std_dev': 1.2195800456608819, 'centroid_stat_min': 4.31960391998291, 'centroid_stat_25%': 5.901429691620457, 'centroid_stat_50%': 6.49275506370837, 'centroid_stat_75%': 7.322224074616932, 'centroid_stat_max': 10.88850212097168}

I am happy to modify it further if you could provide a pandaframe example. Thanks. @krittin-kulrat

**Comment by LilacHo** [comment]
@loriwon-uw Here is how you call them (I don't fully remember but the your input is three parameter?): 
`from data_process import input_file_from_data_dir`
`from geo_process import create_hexagon`
`from height_acquire import height_acquire, get_centroid, average_building_height_with_centroid`

`hex = create_hexagon(longitude, latitude, radius_meters)`

- if seattle (within the boundary defined by seattle_city_limits.geojson):
`building = gpd.read_file(input_file_from_data_dir("seattle_building_footprints.geojson"))`

- if not Seattle: 
`building = height_acquire(hex)`


`new_building = get_centroid(building)`
`average_building_height_with_centroid(new_building, hex)`

The output is as stated in the previous conversation. I will need to adjust it for better link with ML.

**Reply by LilacHo** [reply]
Oh BTW it could take some minutes (I would say 3-5) to run if not Seattle. I first thought it failed because of no response but I ended up finding that it just takes time.
